src/bundle/ui-dev/src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -303,7 +303,7 @@ const UniversalDiscoveryModule = (props) => {
     }, [currentView]);
 
     useEffect(() => {
-        if (!props.startingLocationId || props.startingLocationId === 1) {
+        if (!props.startingLocationId || props.startingLocationId === 1 || props.startingLocationId === props.rootLocationId) {
```

Comment:
You should also update the dependencies of this hook (line :324). 

Dependencies are responsible for establishing what properties will cause the `useEffect` hook to recalculate. In this case you're adding `props.rootLocationId` as part of this condition, so if it ever changes, it should cause `useEffect` to re-apply, just  as `props.startingLocationId` does.

Although, from what I see, other dependencies are also missing (like `currentView`).